### ISSUE TITLE
Expose activation env vars as JSON

### ISSUE THREAD

**Issue by borchero** [prefix-dev/pixi#1298]
## Motivation

In some instances, it is convenient to not require `pixi run` in order to make use of `pixi`-installed binaries (one example might be standalone binaries such as `terraform`). When working locally, this is easily possible by running `eval $(pixi shell-hook)`. However, this might not be very convenient in CI environments such as GitHub Actions.

In this instance, it would be more convenient for tools to get access to the environment variables (and path modifications) that are set by the "activation" of a pixi environment and export them in a way that integrates well with the CI environment. For GitHub Actions, this would be appending to `$GITHUB_PATH` and `$GITHUB_ENV`, allowing all subsequent steps in a workflow job to access `pixi`-installed binaries (even if using an external action that has no notion of the installation process of the binary).

_This feature request is a follow-up from the discussion in https://github.com/prefix-dev/setup-pixi/issues/103._

## Potential Implementation

Given the existing functionalities in `pixi`, the implementation seems rather straightforward. The two main questions would be:

- How should this feature be integrated into the CLI interface?
- What is an appropriate output format?

### CLI Interface

The most straightforward implementation would likely be to add the option to emit JSON as part of the `shell-hook` subcommand. From a naming perspective, this is, however, not ideal (`pixi shell-hook --shell json`?). Instead, one could consider adding a new top-level subcommand (`pixi env`?) although this might be a little prominent given the narrow use case of this feature.

### Output Format

Given the way that GitHub Actions distinguishes between `PATH` and other environment variables, a usable output format would need to supply these separately. For maximum transferability across different tools, JSON seems like an appropriate choice for the format as well.

A potential schema could be:

```json
{
    "version": 1,
    "pathComponents": ["<path>"],
    "environmentVariables": {
        "<key>": "<value>"
    }
}
```

The `pathComponents` should provide the paths that would be added to `PATH` by the environment activation, `environmentVariables` would provide all other new & modified environment variables.

The `version` key is likely optional, however, it would allow for breaking changes to the format without breaking other tools relying on it silently.

**Comment by borchero** [comment]
Once we're aligned on the design, I'm happy to contribute this myself :)

**Comment by ruben-arts** [comment]
He @borchero or @wolfv Do you know of an example where JSON is used in GitHub actions? I'm not familiar with it. 

From a cli design. This seems like a CI only tool so I wouldn't mind it to be behind a flag in a existing sub-command like `pixi shell-hook` as this is not something users would have to type daily. And sub-command "real estate" is expensive :smile: 

Good to hear you want to contribute! I need some more info to understand it completely.

**Comment by borchero** [comment]
> Do you know of an example where JSON is used in GitHub actions

I'm not sure what exactly you mean with that 🤔 since "native" GitHub Actions are written in TS/JS, working with JSON in actions is very natural. In TS code, you then need to add to path/the environment independently, hence, the distinction in above's proposed JSON format:

```typescript
import * as core from '@actions/core'

core.addPath('<path>')
core.exportVariable('<key>', '<value>')
```

Does that already address your point?

> This seems like a CI only tool so I wouldn't mind it to be behind a flag in a existing sub-command like `pixi shell-hook` as this is not something users would have to type daily.

Fair point, `pixi shell-hook -s json` would make most sense then, I guess :smile:

**Comment by ruben-arts** [comment]
Ah I misunderstood the use-case, this would be used in the `setup-pixi` action specifically but it wouldn't be a user focused tool?

I don't have an opinion on the JSON schema, possibly @wolfv or @pavelzw have some input on that.

**Comment by borchero** [comment]
> this would be used in the setup-pixi action specifically but it wouldn't be a user focused tool?

Yes! There might be an application beyond `setup-pixi` but this is the motivation behind this feature.

**Comment by pavelzw** [comment]
IMO this should be an opt-in feature in setup-pixi (I think we all agree on that).
We probably need to handle `PATH` a bit differently than the other environment variables as there is an extra `addPath` command that appends instead of overwrites the variable.

We probably also want to think about how pixi should behave when there are additional things happening in the activation scripts (for example in the MSVC compiler, the compilers are downloaded during activation because of licensing reasons).

This might be actually sort-of related to #1118 as setting environment variables also needs to happen in some way when creating "already activated" docker images.
